cmake/modules/FindSphinx.cmake

```diff
@@ -0,0 +1,18 @@
+#
+# FindSphinx
+# Locate Sphinx-build executable
+#
+# This module will set the following variables when 
+# find_package(Sphinx) is called :
+# 
+# SPHINX_EXECUTABLE : Path to the sphinx-build execuatble
+
+find_program(SPHINX_EXECUTABLE
+            NAMES sphinx-build
+            DOC "Path to the sphinx-build executable") 
+            
+include(FindPackageHandleStandardArgs)
+
+find_package_handle_standard_args(Sphinx
```

Comment:
Should Sphinx and Doxygen be added as an external dependency similar to PLog, ImGui, etc.?  If so, we should add them to the superbuild section for downloading and building the project (if building is necessary) to ease local installation of SmartPeak from source.

Reply:
Doxygen and Sphinx are easily installed in macOS and Linux with package managers, on Windows it's only a matter of downloading the precompiled executables for Doxygen and Sphinx is manager with `pip` and letting CMake know where they are. We could integrate Doxygen in the superbuild but this will be Windows specific as it's not needed on Linux or macOS.

Reply:
Can you add these instructions to the Docs under the sections for building on Windows and also building on MacOS/Linux?

Reply:
I have updated the `README.rst` with instructions on how to fetch the dependencies and generate the docs.